pkg/render/intrusion_detection.go

```diff
@@ -1136,14 +1177,262 @@ func (c *intrusionDetectionComponent) intrusionDetectionAnnotations() map[string
 	return c.cfg.TrustedCertBundle.HashAnnotations()
 }
 
-func (c *intrusionDetectionComponent) intrusionDetectionADJobsPodTemplate() []client.Object {
-	trainingJobPodTemplate := c.getBaseIntrusionDetectionADJobPodTemplate(ADJobPodTemplateBaseName + ".training")
-	detecionADJobPodTemplate := c.getBaseIntrusionDetectionADJobPodTemplate(ADJobPodTemplateBaseName + ".detection")
+// AD API RBAC for accessing token and subject access reviews for AD Pod token verification
+func (c *intrusionDetectionComponent) adAPIServiceAccount() *corev1.ServiceAccount {
+	return &corev1.ServiceAccount{
+		TypeMeta: metav1.TypeMeta{Kind: "ServiceAccount", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ADAPIObjectName,
+			Namespace: IntrusionDetectionNamespace,
+		},
+	}
+}
+
+func (c *intrusionDetectionComponent) adAPIAccessClusterRole() *rbacv1.ClusterRole {
+	return &rbacv1.ClusterRole{
+		TypeMeta: metav1.TypeMeta{Kind: "ClusterRole", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: ADAPIObjectName,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{"authorization.k8s.io"},
+				Resources: []string{"subjectaccessreviews"},
+				Verbs:     []string{"create"},
+			},
+			{
+				APIGroups: []string{"authentication.k8s.io"},
+				Resources: []string{"tokenreviews"},
+				Verbs:     []string{"create"},
+			},
+		},
+	}
+}
+
+func (c *intrusionDetectionComponent) adAPIAccessRoleBinding() *rbacv1.ClusterRoleBinding {
+	return &rbacv1.ClusterRoleBinding{
+		TypeMeta: metav1.TypeMeta{Kind: "ClusterRoleBinding", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: ADAPIObjectName,
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     ADAPIObjectName,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      ADAPIObjectName,
+				Namespace: IntrusionDetectionNamespace,
+			},
+		},
+	}
+}
+
+// AD API Service and Deployment
+func (c *intrusionDetectionComponent) adAPIService() *corev1.Service {
+	return &corev1.Service{
+		TypeMeta: metav1.TypeMeta{Kind: "Service", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ADAPIObjectName,
+			Namespace: IntrusionDetectionNamespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{
+				"k8s-app": ADAPIObjectName,
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Name: ADAPIObjectPortName,
+					Port: adAPIPort,
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: adAPIPort,
+					},
+					Protocol: corev1.ProtocolTCP,
+				},
+			},
+			Type: corev1.ServiceTypeClusterIP,
+		},
+	}
+}
+
+func (c *intrusionDetectionComponent) adAPIDeployment() *appsv1.Deployment {
+	adAPIStorageVolumePath := "/storage"
+	adAPIStorageVolumeName := "volume-storage"
+	envVars := []corev1.EnvVar{
```

Comment:
nit: I do not think there is a reason to put the envvars in a variable as opposed to just copying the lines into the &appsv1.Deployment{}

Reply:
https://github.com/tigera/operator/pull/1931/commits/4546b11522d4ea1557bf00850577c3a07a6522aa